shop/src/Components/ShopBy/index.jsx

```diff
@@ -1,23 +1,72 @@
 import Price from "../Price";
-import CheckboxComponent from "../CheckboxComponent";
+import "./ShopBy.scss";
 import colors from "../../Data/colors.json";
-import {sizes} from "../../Data/sizeAndSortList.js";
+import { sizes } from "../../Data/sizeAndSortList.js";
 import brands from "../../Data/brands.json";
-import "./ShopBy.scss";
-const ShopBy = () => {
+import CheckboxComponent from "../CheckboxComponent";
+import { useState } from "react";
+
+const ShopBy = ({
+  shoeItems,
+  handleOnPriceRange,
+  handleOnGetItemFromCheckbox,
+}) => {
+  const [allCheckedItem, setAllCheckedItem] = useState([]);
+
+  const checkboxResult = shoeItems.filter((shoe) => {
```

Comment:
Amazing 👏.
but please add some comments explaining what this function does and how it work. its a bit complex so if we come back to looks at it after a while those comments will help